superset/views/core.py

```diff
@@ -1279,6 +1279,7 @@ def explore(self, datasource_type=None, datasource_id=None):
         form_data['datasource'] = str(datasource_id) + '__' + datasource_type
 
         # On explore, merge extra filters into the form data
+        form_data = utils.split_adhoc_filters_into_base_filters(form_data)
```

Comment:
Your function mutates the form data so there's no need to re-assign. You can either make `split_adhoc_filters_into_base_filters` create a copy and return the mutated copy or simply mutate the input which seems to be the pattern as per line #243.